server/athenian/api/internal/miners/github/pull_request.py

```diff
@@ -3103,3 +3106,28 @@ def _validate(self, facts: PullRequestFacts, url: str) -> None:
                 facts.released - facts.merged,
             )
             raise ImpossiblePullRequest()
+
+
+async def prs_node_id_to_number(
+    node_ids: Sequence[int],
+    meta_ids: tuple[int, ...],
+    mdb: DatabaseLike,
+) -> dict[int, int]:
+    """Build a mapping to convert PR node identifier to PR number.
+
+    The mapping will not include not found node identifiers.
+    """
+    selects = []
+    node_id_cond, hints = column_values_condition(NodePullRequest.node_id, node_ids)
+    selects = [
+        sa.select(NodePullRequest.node_id, NodePullRequest.number).where(
+            NodePullRequest.acc_id == meta_id, node_id_cond,
+        )
+        for meta_id in meta_ids
+    ]
+    stmt = sa.union_all(*selects)
```

Comment:
```suggestion
    stmt = selects[0] if len(selects) == 1 else sa.union_all(*selects)
```
(this used to generate broken code sometimes, maybe SA fixed it but IDK. anyways, a bit simpler from the logic PoV)

Reply:
yes SA is already doing the expected thing here, so it's a matter of preference